### PR TITLE
Cleaning Code + Simple CLI

### PR DESCRIPTION
**Pull request by guimorg**
## Introduction
Hello Ahmed!

I saw your post on Reddit and I thought about contributing to the project!

First of all, this is awesome! I really enjoyed the project 💯 

## Description

Here are some of my contributions:

- In this PR I am basically refactoring some of the code you wrote into separate packages, that allow a better understanding of the code and better maintainability!
- I implemented a very, very, very simple CLI to run the code, I always prefer running packages from my terminal, so I thought that would be nice!
- Some housekeeping for Python Packages - `pyproject.toml`, `setup.py`, `setup.cfg`. Lately, I've been storing the requirements on the `setup.py` file, instead of using the `requirements.txt` file. Also the name of folders, some required files were created, modified.

## Quality

To improve Code Quality and Maintainability I also included a `.pre-commit.yaml` file to use pre-commit hooks - basically executing flake8 and black, because it is an open-source project I think it is better to use black.

## Things to Watch

I would like to contribute more to the project, I just got my hands on the repo so I didn't have much time to develop more. But I think that we (you) should keep an eye on:

- Documentation: I tried reading a little bit of the algorithm behind the logic but it is a little hard to understand, I think better documentation for the logic/algorithm would be great!
- Tests: we can't get away with unit testing, so it would be great to include some unit testing.
- Performance: I saw you mention that performance was an issue - I kind of agree with that, there should be a threshold for the improvement of fitness (if the fitness does not improve more than X in the last 1000 iterations, stop). If the algorithm is better documented we can also implement a multiprocessing algorithm.
- Checkpoints: I saw that you have implemented a sort of checkpoint - I have maintained the checkpoint saving function, but there is currently no support for reading these checkpoints.

Here is my Monalisa after 50000 iterations using the 64x64 image Monalisa in the `images/` folder
Figure_1

## Ending

Well, any comments are much appreciated! I hope I can contribute to the project 😄!
Guilherme de Amorim
[guimorg](http://guimorg.github.io)

### CODE REVIEW

File: circle_evolution/helpers.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    return target
+
+
+def show_image(img_arr):
+    """Displays image on window.
+
+    Arguments:
+        img_arr: image array to be displayed
+    """
+    plt.imshow(img_arr, cmap="gray", vmin=0, vmax=255)
```
Comment (ahmedkhalf) [inline circle_evolution/helpers.py:44]: add plt.figure() before imshow
```suggestion
    plt.figure()
    plt.imshow(img_arr, cmap="gray", vmin=0, vmax=255)
```

File: circle_evolution/__main__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from main import main
+
+
+if __name__ == "__main__":
+    main()
```
Comment (ahmedkhalf) [inline circle_evolution/__main__.py:5]: What exactly is the use of this file?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    packages=find_packages(),
+    python_requires=">=3.6",
+    setup_requires=["wheel"],
+    entry_points={
+        "console_scripts": ["circle_evolution=circle_evolution.main:main"]
+    },
+    install_requires=INSTALL_REQUIRES,
+    extras_require={"test": TEST_REQUIRES},
+    classifiers=[
+        "Environment :: Web Environment",
```
Comment (ahmedkhalf) [inline setup.py]: Why is the environment Web Environment here? Am I missing something?

**Review by ahmedkhalf — CHANGES_REQUESTED** [review]
Thank you for your contribution, it is greatly appreciated. I just reviewed the code, and everything is looking good overall. I added two comments to your code. Also please review my change request on the file setup.py
Also to have a more direct line of communication please add me on discord Itwist#7732, so I can invite you to the dev group.
Many Thanks, Ahmed Khalf.

File: setup.py (hunk 3 above)
Comment (guimorg) [reply setup.py]: Oops! 😬
Traits of a personal `setup.py` file I use for reference! Removed

**Comment by ahmedkhalf** [comment]
Merged! @guimorg Please add me on discord Itwist#7732 so we can have a more direct line of communication.

**Pull request merged**